### PR TITLE
PyPi integration and pyreadr python only version.

### PR DESCRIPTION
**Pull request by airqualityanthony**
Hi, 

I’ve tried to make some improvements to this by making it a python only integration of the function, and I’ve also packaged it up and released for pip install on pypi to aid end user installation. 

I don’t think keeping a fork open is the best method to continue running my downstream version. I can try this pull request, though I’m unsure on whether the actions and releases will carry over (I integrated an action to automatically upload new release to PyPi). 

I was planning on making further improvements, so if you’re happy to maintain and run the master branch handling the pull requests I’ll leave it at that. Otherwise I think you can transfer ownership of repositories over, though understand if you don’t want to do that. 

Thanks,

Anthony

### CODE REVIEW

**Comment by robintw** [comment]
Hi,

Thanks for these alterations. Getting rid of the dependency on R is great!

I'm fine with this being on PyPI, but in future it's probably best to ask the original author before publishing a fork of a package on PyPI. Particularly as in this case I hadn't put a license on the code when I originally uploaded it (which was an oversight on my part) and you've added a MIT license with your name on it. I don't really mind much, but there are some original authors in my position who might be rather cross about this. Submitting the PR before running the pipeline to upload to PyPI would have been fine.

Anyway, I'm happy to merge this, but could you update the copyright line in the MIT license file to include my name too?

I'm very happy to accept future modifications - really glad someone is finding this useful.

**Comment by airqualityanthony** [comment]
Hi Robin,

Ahh yeah no worries, I'm always a little confused with open source etiquette on what can be done etc, it's my first time approaching pypi packaging and I got a little carried away learning and coding it up! Thanks for clarifying and apologies for not running it by yourself first! 

I had added you to the pypi package author list (not sure if that informed you), but oversight on the MIT doc. Have updated with latest commit! 

Indeed, myself and a few colleagues have recently been lamenting all the great python data science work and having to bridge the air quality data with R! Hopefully I can continue to contribute more, initial idea is just to continue to port openair functionality. 

I'll keep an eye on the action workflow for pypi when pull request accepted. It's only triggered by a release submission, which I think can be handled by pull requests from my side! 

Thanks,

Anthony

**Comment by airqualityanthony** [comment]
Oh and also can add you to the pypi package as an owner as well if you’d like! Just let me know a pypi username!

**Comment by robintw** [comment]
Thanks for the updates.

I'm happy to merge this now (and then we get a new release), or I can keep it open if you're planning to do more changes very shortly and want to keep it all together - I'm happy either way.

Yes, I'd be happy to be a PyPI package owner: my username there should be `robintw`

**Comment by airqualityanthony** [comment]
Added you as owner! 

Happy to merge now :)

File: src/pyaurn.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        statistical method to apply to group values. ["mean","max","min","median","sum"]
+
+    Returns
+    ----------
+    pandas.DataFrame
+        returns a pandas.DataFrame time averaged to specified frequency & statistic.
+    """
+    if avg_time == "daily":
+        if statistic=="mean":
+            time_df = df.groupby(pd.Grouper(freq='D')).mean()
```
Comment (robintw) [inline src/pyaurn.py]: Just noticed these if statements here - you can probably make this a lot simpler by using the `agg` method in Pandas. That does an aggregation, and can take a string giving the name of the aggregation to do.

So you can just do `df.groupby(pd.Grouper(freq='D').agg(statistic)`

**Comment by robintw** [comment]
Great. Just looked at the code and added one comment that could simplify some of your aggregation code.

File: src/pyaurn.py (hunk 1 above)
Comment (airqualityanthony) [reply src/pyaurn.py]: God thank you, I was trying to concat the statistic variable and parentheses. Will update now.

**Comment by airqualityanthony** [comment]
Thanks, I knew there must be a better way. Was trying to make it work with trying to concatenate the statistic with parentheses.

Should be all good now!

**Comment by airqualityanthony** [comment]
Hi Robin,

Ahh yeah no worries, I'm always a little confused with open source
etiquette on what can be done etc, it's my first time approaching pypi
packaging and I got a little carried away learning and coding it up! Thanks
for clarifying and apologies for not running it by yourself first!

I had added you to the pypi package author list (not sure if that informed
you), but oversight on the MIT doc. Will update!

Indeed, myself and a few colleagues have recently been lamenting all the
great python data science work and having to bridge the air quality data
with R! Hopefully I can continue to contribute more, initial idea is just
to continue to port openair functionality.

I'll keep an eye on the action workflow for pypi when pull request
accepted. It's only triggered by a release submission, which I think can be
handled by pull requests from my side!

Thanks,

Anthony

**Pull request merged**